### PR TITLE
Fix inconsistent aggregate function states in mixed x86-64 / ARM clusters

### PR DESCRIPTION
**Pull request by HarryLeeIBM**
In mixed x86-64 / ARM clusters, aggregate function states can be created differently since ARM sometimes calculates slightly different floating-point results.

This [blog](https://altinity.com/blog/announcing-arm64-altinity-stable-builds-for-clickhouse) contains some related tests for the issue.

The fix is to use compiler flag `-ffp-contract=off` to make sure ARM floating-point calculation produce the same results as on x86-64.

The functional test `02813_seriesDecomposeSTL.sql` (which was disabled on ARM) can be used to test the fix, this PR enables it on ARM.

### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fixed potentially inconsistent aggregate function states in mixed x86-64 / ARM clusters.

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit 51873f0dc9d45dc6993ffdd8379727761c168cb2 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/60610/51873f0dc9d45dc6993ffdd8379727761c168cb2/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

**Comment by HarryLeeIBM** [comment]
@rschu1ze, could you or somebody in ClickHouse team review the PR? Thanks!

**Comment by rschu1ze** [comment]
Besides that, test results in CI look good.

File: CMakeLists.txt [hunk 1]
```diff
@@ -319,8 +319,8 @@ if (COMPILER_CLANG)
     endif()
 endif ()
 
-set (COMPILER_FLAGS "${COMPILER_FLAGS}")
-
+# Disable floating-point expression contraction in order to get consistent floating point calculation results across platforms
+set (COMPILER_FLAGS "${COMPILER_FLAGS} -ffp-contract=off")
```
Comment (rschu1ze) [inline CMakeLists.txt:323]: Please put in a newline after l. 323 to make this visually more pleasing, and I'll be happy :-)

**Comment by rschu1ze** [comment]
Performance of queries with functions `arrayReduce` and `arrayReduceInRanges` deteriorated. Let's re-run CI and check if the problem persists/

File: CMakeLists.txt (hunk 1 above)
Comment (HarryLeeIBM) [reply CMakeLists.txt:323]: Done

**Comment by rschu1ze** [comment]
ClickHouse Integration Tests (tsan) [6/6]:
- test_storage_kafka/test.py::test_system_kafka_consumers_rebalance_mv #60443 (was fixed in the meantime)

Performance tests:
- one run still failed due to unrelated issues (https://s3.amazonaws.com/clickhouse-test-reports/60610/3a4ef70a208ca4452f646bb78f1fb96a12bc377e/performance_comparison_[2_4]/report.html)
- https://s3.amazonaws.com/clickhouse-test-reports/60610/3a4ef70a208ca4452f646bb78f1fb96a12bc377e/performance_comparison_[4_4]/report.html has two queries on x86 which is surprising given that `-ffp-contract=off` was already the default on x86 (AFAIK). But looking at the queries, I don't think the performance regression is related anyways.

ClickHouse AST Fuzzer fail with a generic error. I looked at the server and fuzzer logs, but it is not really clear to me what is going wrong. For sure no sanitizer issue and unrelated.

**Pull request merged**